### PR TITLE
Brian/round bid quantity

### PR DESCRIPTION
**Pull request by bweick**

### CODE REVIEW

File: contracts/core/modules/RebalanceAuctionModule.sol [hunk 1]
```diff
@@ -76,11 +76,24 @@ contract RebalanceAuctionModule is
      */
     function bid(
         address _rebalancingSetToken,
-        uint256 _quantity
+        uint256 _quantity,
+        bool _executePartialQuantity
```
Comment (asoong) [inline contracts/core/modules/RebalanceAuctionModule.sol]: I think you had the naming right in the comment below. Maybe `allowPartialFill` or `shouldPartiallyFill`?

File: test/contracts/core/tokens/rebalancingSetToken.spec.ts [hunk 2]
```diff
@@ -620,7 +620,8 @@ contract('RebalancingSetToken', accounts => {
         const [minimumBid] = await rebalancingSetToken.getBiddingParameters.callAsync();
         await rebalanceAuctionModule.bid.sendTransactionAsync(
           rebalancingSetToken.address,
-          minimumBid
+          minimumBid,
+          false
```
Comment (asoong) [inline test/contracts/core/tokens/rebalancingSetToken.spec.ts]: I noticed this when I was going in to update the specs to use the deployed contracts. Why don't you specify the transaction data when you send transactions?

File: test/contracts/core/tokens/rebalancingSetToken.spec.ts [hunk 3]
```diff
@@ -755,7 +756,8 @@ contract('RebalancingSetToken', accounts => {
         const minimumBid = biddingParameters[0];
         await rebalanceAuctionModule.bid.sendTransactionAsync(
           rebalancingSetToken.address,
-          minimumBid
+          minimumBid,
+          false
```
Comment (asoong) [inline test/contracts/core/tokens/rebalancingSetToken.spec.ts]: This block between `756-760` is repeated over and over again. Consider adding a rebalancing helper method that also defaults the partial fill parameter to false

File: contracts/core/modules/RebalanceAuctionModule.sol (hunk 1 above)
Comment (felix2feng) [inline contracts/core/modules/RebalanceAuctionModule.sol]: Add javadoc

File: contracts/core/modules/RebalanceAuctionModule.sol [hunk 4]
```diff
@@ -76,11 +76,24 @@ contract RebalanceAuctionModule is
...
     )
         external
         nonReentrant
     {
+        // If user allows partial fills calculate partial fill (if necessary)
+        uint256 executionQuantity;
```
Comment (felix2feng) [inline contracts/core/modules/RebalanceAuctionModule.sol]: Why not use a ternary operator?
```
executionQuantity = _executePartialQuantity ?
   calculateExecutionQuantity( _rebalancingSetToken, _quantity):
   _quantity;
```

File: contracts/core/modules/RebalanceAuctionModule.sol [hunk 5]
```diff
@@ -126,11 +139,24 @@ contract RebalanceAuctionModule is
...
+        if (_executePartialQuantity) {
```
Comment (felix2feng) [inline contracts/core/modules/RebalanceAuctionModule.sol]: We can consolidate this logic into a function like `getExecutionQuantity(_quantity, _executePartialQuantity)` that just returns the execution quantity.

File: contracts/core/modules/RebalanceAuctionModule.sol [hunk 6]
```diff
@@ -224,7 +250,39 @@ contract RebalanceAuctionModule is
...
+        uint256 minimumBid = biddingParameters[0];
+        uint256 remainingCurrentSets = biddingParameters[1];
+
+        // If quantity is greater than remainingCurrentSets round amount to nearest multiple of
+        // minimumBid that is less than remainingCurrentSets
+        if (_quantity <= remainingCurrentSets) {
+            return _quantity;
+        } else {
+            // Find amount that is nearest multiple of minimumBid less than remainingCurrentSets
+            uint256 executionQuantity = remainingCurrentSets.div(minimumBid).mul(minimumBid);
```
Comment (felix2feng) [inline contracts/core/modules/RebalanceAuctionModule.sol]: What if the quantity is less than the minimum bid? Then it's 0?

File: contracts/core/modules/RebalanceAuctionModule.sol [hunk 7]
```diff
@@ -224,7 +250,39 @@ contract RebalanceAuctionModule is
...
 
-    /* ============ Public Functions ============ */
+    /* ============ Internal Functions ============ */
+
+    /**
+     * Get execution quantity in event bid quantity exceeds remainingCurrentSets
+     *
+     * @param  _rebalancingSetToken    Address of the rebalancing token being bid on
+     * @param  _quantity               Number of currentSets to rebalance
+     * @return executionQuantity       Array of token addresses invovled in rebalancing
```
Comment (felix2feng) [inline contracts/core/modules/RebalanceAuctionModule.sol]: Fix javadoc

File: contracts/core/tokens/RebalancingSetToken.sol [hunk 8]
```diff
@@ -318,22 +318,22 @@ contract RebalancingSetToken is
         external
         returns (address[] memory, uint256[] memory, uint256[] memory)
     {
+        // Validate bid quantity and module is sender
+        StandardPlaceBidLibrary.validateBidQuantity(
```
Comment (felix2feng) [inline contracts/core/tokens/RebalancingSetToken.sol]: 💯

**Pull request merged**